parsepatch/patch.py

```diff
@@ -88,6 +88,15 @@ def parse_file(filename, file_filter=None, skip_comments=True,
                    add_lines_for_new=True):
         with open(filename, 'r') as In:
             patch = In.read()
+
```

Comment:
What is exactly the issue you want to fix ?
i) we could have the same issue with a patch coming from a net stream
ii) maybe we could skip this kind of line in _lines (when matching "^[\r\n]*$" for example) and then handle all the possible cases

Reply:
So if you could add a failing test without this patch and then which will success with it, then it'd be better.

Reply:
I issue is that the final hunk parsed will contain an extra newline. The tests that I wrote will/have failed because there is an extra newline in the final hunk. 
If we remove the newline here we know that it is an empty newline whereas if we do it on a line by line basis they could be part of the hunk and should not be removed.

Reply:
afaik, lines in hunks should begin with a '+' or '-' of ' ' so if a line is matching ^[\r\n]*$ it shouldn't be a part of the hunk.

Reply:
I don't think using '^[\r\n]*$' will work because the final newline must be removed before 'patch.split('\n')'. The line conditions are checked after the patch has already been split.

Reply:
you right, so ^[\r]*$ should be ok for lines containing \r or being empty

Reply:
I've added an additional function skip_newlines(self) which uses the regex you suggested.